doc/examples/applications/plot_3d_structure_tensor.py

```diff
@@ -0,0 +1,160 @@
+# To view the sample data in 3D, run the following code:
+#
+# .. code-block:: python
+#
+#     import plotly.graph_objects as go
+#
+#     (n_Z, n_Y, n_X) = sample.shape
+#     Z, Y, X = np.mgrid[:n_Z, :n_Y, :n_X]
+#
+#     fig = go.Figure(
+#         data=go.Volume(
+#             x=X.flatten(),
+#             y=Y.flatten(),
+#             z=Z.flatten(),
+#             value=sample.flatten(),
+#             opacity=0.5,
+#             slices_z=dict(show=True, locations=[4])
+#         )
+#     )
+#     fig.show()
+
+#####################################################################
+# Compute structure tensor
+# ========================
+# About the brightest region (i.e., at X ~ 22 and Y ~ 17), we can see
+# variations (and, hence, strong gradients) along either X or Y over a typical
+# length of 3.
+
+px.imshow(
+    sample[0, :, :],
+    zmin=v_min,
+    zmax=v_max,
+    labels={'x': 'Y', 'y': 'X', 'color': 'intensity'},
+    title='Interactive view of bottom slice of sample data.'
+)
+
+#####################################################################
+# Therefore, we choose a 'width' of 3 for the window function.
+
+sigma = 3
+A_elems = feature.structure_tensor(sample, sigma=sigma)
+
+#####################################################################
+# We can then compute the eigenvalues of the structure tensor.
+
+eigen = feature.structure_tensor_eigenvalues(A_elems)
+eigen.shape
+
+#####################################################################
+# Where is the largest eigenvalue?
+
+coords = np.unravel_index(eigen.argmax(), eigen.shape)
+assert coords[0] == 0  # by definition
+coords
+
+#####################################################################
+# We are looking at a local property. Let us consider a tiny neighbourhood
+# of this maximum in the X-Y plane.
+
+eigen[0, coords[1], 20:23, 12:14]
+
+#####################################################################
+# If we examine the second-largest eigenvalues in this neighbourhood, we can
+# see that they have the same order of magnitude as the largest ones.
+
+eigen[1, coords[1], 20:23, 12:14]
+
+#####################################################################
+# In contrast, the third-largest eigenvalues are one order of magnitude
+# smaller.
+
+eigen[2, coords[1], 20:23, 12:14]
+
+#####################################################################
+# As expected, the region about voxel ``(Z, Y, X) = coords[1:]`` is markedly
+# anisotropic in 3D: There is an order of magnitude between the third-largest
+# eigenvalues on one hand, and the largest and second-largest eigenvalues on
+# the other hand.
+# This region is 'somewhat isotropic' in the X-Y plane: There is a factor of
+# (only) ~3 between the second-largest and largest eigenvalues.
+# This is definitely compatible with what we are seeing in the image, i.e., a
+# stronger gradient roughly along X and a weaker gradient perpendicular to it.
```

Comment:
This sentence sounds kind of like we are implying the entries in `eigvals` correspond to X,Y, Z axis, but that isn't really the case. From just the `eigvals` array, we don't know the directions of the corresponding eigenvectors.

Reply:
Absolutely. Fixed in 521d4d235e86ec64e709e586fc864850b45044fb. Should we draw the gradient directions on the image? As in, roughly, manually, with a red marker?